libs/state/src/lib/download/download.state.ts

```diff
@@ -0,0 +1,149 @@
+import { Action, State, StateContext } from '@ngxs/store';
+import { Download } from './download.action';
+import { DownloadModel, FileFormat } from './download.model';
+
+//TODO: For testing purpose only
+/**
+ * this is file url which will be picked from
+ * other state later
+ */
+const fileUrl = 'assets/compass.svg';
+
+/**
+ * Download State Model used to convert
+ * file from SVG file content to different
+ * format and download to user.
+ */
+@State<DownloadModel>({
+  name: 'download',
+  defaults: [
+    {
+      format: FileFormat.PDF,
+      label: 'Download PDF',
+    },
+  ],
+})
+export class DownloadState {
+  /**
+   * Download converted formatted file to browser
+   * @param blob
+   * @param fileName
+   */
+  blobDownload(blob: Blob, fileName: string) {
+    const url = window.URL.createObjectURL(blob);
+    const anchor = document.createElement('a');
+    document.body.appendChild(anchor);
+    anchor.download = fileName;
+    anchor.href = url;
+    anchor.click();
+    anchor.remove();
+    window.URL.revokeObjectURL(url);
+  }
+
+  /**
+   * Infers filename from file url
+   * @param fileUrl
+   * @returns
+   */
+  inferFilename(fileUrl: string) {
```

Comment:
Rename to `guessFilename`
Also it doesn't work as intended. Try the following examples:
`abc`
`abc/def`
`abc/def/`